data/preprocess/preprocess_default.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+"""Preprocess a KGE dataset into a the format expected by libkge.
+
+Call as `preprocess.py --folder <name>`. The original dataset should be stored in
+subfolder `name` and have files "train.txt", "valid.txt", and "test.txt". Each file
+contains one SPO triple per line, separated by tabs.
+
+During preprocessing, each distinct entity name and each distinct relation name
+is assigned an index (dense). The index-to-object mapping is stored in files
+"entity_ids.del" and "relation_ids.del", resp. The triples (as indexes) are stored in
+files "train.del", "valid.del", and "test.del". Additionally, the splits
+"train_sample.del" (a random subset of train) and "valid_without_unseen.del" and
+"test_without_unseen.del" are stored. The "test/valid_without_unseen.del" files are
+subsets of "valid.del" and "test.del" resp. where all triples containing entities
+or relations not existing in "train.del" have been filtered out.
+
+Metadata information is stored in a file "dataset.yaml".
+
+"""
+
+import argparse
+import yaml
+import os.path
+import numpy as np
+from collections import OrderedDict
+
+from util import analyze_raw_splits
+from util import store_map
+from util import RawDataset
+from util import RawSplit
+from util import write_dataset_config
+from util import process_splits
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("folder", type=str)
+    parser.add_argument("--order_sop", action="store_true")
```

Comment:
Would be nice to treat this generic, this can just be a simple mapping {0:S, 1:O, 2:P}